internal/dispatch/singleflight/singleflight_test.go

```diff
@@ -187,29 +187,35 @@ func TestSingleFlightDispatcherCancelation(t *testing.T) {
 	disp := New(mockDispatcher{f: f}, &keys.DirectKeyHandler{})
 	wg := sync.WaitGroup{}
 	wg.Add(3)
+	errs := make(chan error, 3)
```

Comment:
Don't understand the reason for the change on this test, but it should be equivalent.

Reply:
This stems from `go-require: require must only be used in the goroutine running the test function (testifylint)`. Most of the lint issues in this PR are a result of using require within goroutines.

Some context on why: `vet` warns on using require within goroutines (`require` is the same as `t.FailNow()` and [this](https://go.dev/doc/go1.16#vet-testing-T) is explicitly [discouraged](https://github.com/stretchr/testify/issues/772#issuecomment-945166599) within a goroutine)

To address this, I used the a channel to collect them and then check after the goroutines have completed.